src/lib/integration_test_cloud_engine/kubernetes_network.ml

```diff
@@ -42,9 +42,9 @@ module Node = struct
     ; mutable should_be_running : bool
     }
 
-  let id { pod_ids; _ } = List.hd_exn pod_ids
+  let id { app_id; _ } = app_id
```

Comment:
Functionally, the id should be the name of the application, not the name of the specific kubectl pod.  the pod id is something that could change (at least once i've merged the node down recovery logic), and that's represented by infra_id.  I wanted to change that to more future proof this api for when we use engines that aren't kubernetes.

There could theoretically be multiple pods, and even though we only use one, i never felt right about just calling List.hd_exn to just grab whatever happened to be the first, seems like a possibility for instability/non determinacy.

Reply:
I've personally found the pod ID much more helpful for debugging. I see the conceptual benefit, but, since this removes information that I know I've used in past, I'd prefer to revert this.

Reply:
i'd rather not change the API, but i can go around to all the places where it's printed out and use Node.infra_id instead which would retain the debug friendly behavior you're used to

Reply:
That would be good, yes please.